storm-client/src/jvm/org/apache/storm/tuple/Values.java

```diff
@@ -23,9 +23,13 @@ public Values() {
     }
 
     public Values(Object... vals) {
-        super(vals.length);
-        for (Object o : vals) {
-            add(o);
+        super(vals != null ? vals.length : 0);
+        if (vals != null && vals.length != 0 ) {
```

Comment:
`vals.length != 0` should be removed in this condition, cause condition for else statement would be `vals == null || vals.length == 0`, which `Values(List())` goes into else statement and comes out of `Values(null)` instead of no element.